pkg/ingester/ingester_test.go

```diff
@@ -4580,6 +4580,10 @@ func prepareIngesterWithBlocksStorageAndLimits(t testing.TB, ingesterCfg Config,
 }
 
 func prepareIngesterWithBlockStorageAndOverrides(t testing.TB, ingesterCfg Config, overrides *validation.Overrides, ingestersRing ring.ReadRing, dataDir string, bucketDir string, registerer prometheus.Registerer) (*Ingester, error) {
+	return prepareIngesterWithBlockStorageAndOverridesAndPartitionRing(t, ingesterCfg, overrides, ingestersRing, nil, dataDir, bucketDir, registerer)
+}
+
+func prepareIngesterWithBlockStorageAndOverridesAndPartitionRing(t testing.TB, ingesterCfg Config, overrides *validation.Overrides, ingestersRing ring.ReadRing, partitionsRing *ring.PartitionRingWatcher, dataDir string, bucketDir string, registerer prometheus.Registerer) (*Ingester, error) {
```

Comment:
Have you considered using `createTestIngesterWithIngestStorage()`. If there's no hard blocker I would prefer to use an ingester which have been fully setup to use the ingest storage (e.g. fake Kafka, etc...).  I think it will also simplify your setup in `TestOwnedSeriesServiceWithPartitionsRing`.

Reply:
Maybe it's not easy to restart the ingester in the "new user trigger from WAL replay" test case, but I would give it a try anyway.

Reply:
Gave that a try. Not sure if it's simpler, but I've realized that I should push series via Kafka, and not `Push` method.